### ISSUE TITLE
installed mode tests fail for non-root user on Solaris

### ISSUE THREAD

**Issue by alanc** [mesonbuild/meson#5902]
```
======================================================================
FAIL: test_installed_modes (__main__.LinuxlikeTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_unittests.py", line 4579, in test_installed_modes
    self.assertEqual(want_mode, found_mode[1:])
AssertionError: 'rw------T' != 'rw-------'
- rw------T
?         ^
+ rw-------
?         ^
```

```
======================================================================
FAIL: test_installed_modes_extended (__main__.LinuxlikeTests)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "run_unittests.py", line 4650, in test_installed_modes_extended
    (fsobj, want_mode, found_mode)))
AssertionError: '-r--r--r-T' != '-r--r--r--'
- -r--r--r-T
?          ^
+ -r--r--r--
?          ^
 : Expected file include/rootdir.h to have mode -r--r--r-T but found -r--r--r-- instead.

```

This appears to be because [Solaris chmod(2)](https://docs.oracle.com/cd/E88353_01/html/E37841/chmod-2.html) has an additional restriction:

>        If the process is not a privileged process and the file is not a direc-
>        tory, mode bit 01000 (S_ISVTX, the sticky bit) is cleared.

**Comment by dcbaker** [comment]
I've seen similar issues with the sticky bit on at least FreeBSD (probably all of the BSDs).

**Comment by eli-schwartz** [comment]
I've been fighting this while trying to get CI running on freebsd, and my conclusion is that it's quite odd to be setting this at all. Anyone who is trying to set this bit on installed files/directories ***clearly*** only wants it for directories, because it's actually useless on files even for OSes where that isn't actively removed. And on freebsd, the problem is that it literally triggers "illegal operation" (EFTYPE) when you try, rather than silently resetting it behind your back.

So I think meson should just proactively remove that mode from anything that isfile instead of isdir.

Making this entirely more complicated is the fact that meson doesn't actually respect install_mode for the top-level directory at all, so you cannot actually set the sticky bit for directories.

In other words, we should rip out the "t/T" feature of install_mode entirely and raise configure-time warnings if you try it because it's obviously incorrect behavior. :/ Literally unfixable.
